src/Core/Models/GraphQLFilterParsers.cs

```diff
@@ -232,6 +251,88 @@ public GQLFilterParser(RuntimeConfigProvider runtimeConfigProvider, IMetadataPro
         return MakeChainPredicate(predicates, PredicateOperation.AND);
     }
 
+    /// <summary>
+    /// For CosmosDB, a nested filter represents a join between
+    /// the parent entity being filtered and the related entity representing the
+    /// non-scalar filter input. This function:
+    /// 1. Recursively parses any more(possibly nested) filters.
+    /// 2. Adds join predicates between the related entities.
+    /// 3. Adds the subquery to the existing list of predicates.
+    /// </summary>
+    /// <param name="ctx">The middleware context.</param>
+    /// <param name="filterField">The nested filter field.</param>
+    /// <param name="subfields">The subfields of the nested filter.</param>
+    /// <param name="columnName">Current Column Name</param>
+    /// <param name="entityType">Current Entity Type</param>
+    /// <param name="predicates">The predicates parsed so far.</param>
+    /// <param name="queryStructure">The query structure of the entity being filtered.</param>
+    private void HandleNestedFilterForCosmos(
+        IMiddlewareContext ctx,
+        IInputField filterField,
+        List<ObjectFieldNode> subfields,
+        string columnName,
+        string entityType,
+        List<PredicateOperand> predicates,
+        CosmosQueryStructure queryStructure,
+        ISqlMetadataProvider metadataProvider)
+    {
+        string entityName = metadataProvider.GetEntityName(entityType);
+
+        // Validate that the field referenced in the nested input filter can be accessed.
+        bool entityAccessPermitted = queryStructure.AuthorizationResolver.AreRoleAndOperationDefinedForEntity(
+            entityIdentifier: entityName,
```

Comment:
Here the permission checking is using the "EntityName", in the caller of ```HandleNestedFilterForCosmos```, it's getting the parameter value by using ```string nestedFieldTypeName = fieldDefinitionNode.Type.NamedType().Name.Value;```  Last time that I remember, Cosmos allows to have a different GraphQL type name than the entity name in the config. and it uses the `model` directive to map between the two. So if the name originally provided doesn't match any entity name, will this permission check always fail?

Reply:
@tarazou9 if you are talking about aliasing..
here is the config https://github.com/Azure/data-api-builder/blob/9da650e6b4f5545349935247aa84e58fc58e88d2/src/Service.Tests/dab-config.CosmosDb_NoSql.json#L429

Here is the test cases covering that: https://github.com/Azure/data-api-builder/blob/9da650e6b4f5545349935247aa84e58fc58e88d2/src/Service.Tests/CosmosTests/QueryFilterTests.cs#L161